### PR TITLE
Added OpenGL3 support

### PR DESCRIPTION
**Pull request by Jakobzs**
Still very WIP but open for review now 😄

### CODE REVIEW

File: hudhook/src/hooks/dx9.rs
Comment (veeenu) [inline hudhook/src/hooks/dx9.rs:63]: @FrankvdStam I haven't tested DX9 much; do you think this is appropriate here, in the same way as DX11/12? Looks good to me but not sure if you left it out on purpose.

File: hudhook/src/hooks/opengl3.rs
Comment (veeenu) [inline hudhook/src/hooks/opengl3.rs:218]: I'd rather remove this and have warnings if the `RawDetour` is actually unused -- which it shouldn't anyway since it's in `unhook()`.

File: hudhook/src/hooks/opengl3.rs
Comment (veeenu) [inline hudhook/src/hooks/opengl3.rs:217]: I'd rename this as `OpenGl3Hooks` to comply with the [API guidelines](https://rust-lang.github.io/api-guidelines/naming.html) that state:

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use `Uuid` rather than `UUID`, `Usize` rather than `USize` or `Stdin` rather than `StdIn`.

Chances are this is a refactor that has to be done elsewhere in the codebase too, but it's worth it to do it as we go.

File: hudhook/src/hooks/opengl3.rs
Comment (veeenu) [inline hudhook/src/hooks/opengl3.rs:273]: Can we move this before the `OpenGL3Hooks` struct definition? I'd prefer having dependent functions below dependencies.

File: imgui-opengl3/src/lib.rs
Comment (veeenu) [inline imgui-opengl3/src/lib.rs:3]: This is perfectly ok if it works for current use cases, but yeah, I'd eventually like to embed our own implementations if possible.

**Review by veeenu — CHANGES_REQUESTED** [review]
Thank you for this PR, there are just a few minor style concerns to address but it overall looks good to me!

File: hudhook/src/hooks/common.rs
Comment (Jakobzs) [reply hudhook/src/hooks/common.rs:10]: Renamed to ImguiOpenGl3Hooks (due to the comment below about renaming OpenGL3Hooks)

File: hudhook/src/hooks/dx9.rs
Comment (Jakobzs) [reply hudhook/src/hooks/dx9.rs:63]: Awaiting reply here is probably the best yep

File: hudhook/src/hooks/opengl3.rs
Comment (Jakobzs) [reply hudhook/src/hooks/opengl3.rs:217]: Renamed

File: hudhook/src/hooks/opengl3.rs
Comment (Jakobzs) [reply hudhook/src/hooks/opengl3.rs:273]: Done

File: imgui-opengl3/src/lib.rs
Comment (Jakobzs) [reply imgui-opengl3/src/lib.rs:3]: Agreed :)

**Review by veeenu — APPROVED** [review]
Waiting on feedback wrt `setup_io` but otherwise LGTM!

File: hudhook/src/hooks/dx9.rs
Comment (FrankvdStam) [reply hudhook/src/hooks/dx9.rs:63]: Nice seeing an openGL target being added, nice work Jakobzs!
As for this line of code, I'm honestly not sure, I still didn't find the time to actually use the dx9 target in a real application. Did you actually try, and had problems, leading to you adding this line?

Comment (Jakobzs) [reply hudhook/src/hooks/dx9.rs:63]: Thank you :D

I have honestly not tested it on a DX9 target as I do not know one currently (all I know are DX11 and up). But from reasoning of that it was an issue in this hook (hence the added code), and also I based a lot of it on yours, then I assumed it might have been an issue there as well. It is a bit out of scope of this pull request tbh, but could be a nice fix if it is really needed.

Comment (FrankvdStam) [reply hudhook/src/hooks/dx9.rs:63]: If these are window events like resizing imgui windows, then that is already working without this.  There's at least 1 bug that I know of, which is that the game will crash when changing the resolution. Not sure if this could be related.

Maybe its better to just leave this alone until one of us actually uses it in a dx9 game, thats' when we'll find all the bugs.

Comment (Jakobzs) [reply hudhook/src/hooks/dx9.rs:63]: Could probably be ideal yeah. I will just revert it.

**Pull request merged**
